designs/autoscaler.md

```diff
@@ -0,0 +1,515 @@
+
+Cluster Autoscaler provides the flexibility to enable autoscaling, whether it is within the cluster, or for a remote workload cluster. Cluster Autoscaler can be:
+
+* deployed in the management cluster to autoscale the management cluster itself
+* deployed in the management cluster to autoscale a remote workload cluster
+* deployed in the workload cluster to autoscale the workload cluster itself
+
+Since EKS Anywhere follows a standard management/workload clusters setup where all the CAPI machine objects are defined in the management cluster, we choose to **deploy autoscalers in the management cluster as well — to watch and manage management cluster, and all the remote workload clusters, as default option**. In this way, we have a centralized place — management cluster, to deploy and manage all the autoscaler controllers for workload clusters. 
+
+However, it is not uncommon that the management cluster size defined by the user is small, but the workload clusters are rather big, then it makes more sense to distribute the workload and run the autoscaler in each workload cluster to reduce resource usage in one management cluster. This configuration is also supported by us. Though this approach requires extra steps to create the management cluster’s `kubeconfig` in the workload cluster, comparing to the first approach where each workload cluster’s kubeconfig by default exists in the management cluster. Having management cluster kubeconfig in workload cluster also draws security concerns. 
+
+EKS Anywhere user can either deploy Cluster Autoscaler controller themselves to target cluster by applying its [deployment manifest](https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/clusterapi/examples/deployment.yaml) with custom arguments or using [Helm chart](https://github.com/kubernetes/autoscaler/tree/master/charts/cluster-autoscaler), or easy-install the Cluster Autoscaler as EKS-A curated package which is Amazon built, scanned, and tested. 
+
+### EKS-A Curated Package
+
+We will build the Cluster Autoscaler as an EKS-A curated package with an opinionated package bundle with default values. Once built and included in the package list, the autoscaler installation should fit into the standard curated package workflow. 
+
+## User Experience
+
+There are two requirements to enable cluster autoscaling:
+
+1. annotate the CAPI MachineDeployment — configurable through EKS-A cluster spec
+2. install Cluster Autoscaler controller in the cluster
+
+Although it is not required, we recommend that users install the autoscaler as EKS-A curated package through the EKS-A CLI for a more streamlined installation experience and better support. There are a few possible scenarios when enabling autoscaling. 
+
+**Enable Cluster Autoscaler as a Curated Package During Cluster Creation**
+
+First, define a cluster spec file with autoscaling config enabled for selected worker node group(s):
+
+```yaml
```

Comment:
I guess if the min/max was in the package configuration, it would also need this machine group name.

Reply:
That's another reason I don't think autoscaler config should be part of package config. We need to map the worker node group and min/max together. It creates dependencies to eks-a main workflow/cluster spec. Plus the reason https://github.com/aws/eks-anywhere/pull/2630#discussion_r913893248. I feel we should have autoscaler config as part of the cluster spec.